integration/pages/RosterTestPage.js

```diff
@@ -0,0 +1,155 @@
+const BaseTestPage = require('./BaseTestPage');
+const assert = require('assert');
+const { By } = require('selenium-webdriver');
+
+const ELEMENTS = {
+  toggleMicrophoneButton: By.css(
+    "[data-testid='button'][label='Mute'], [data-testid='button'][label='Unmute']"
+  ),
+  toggleVideoButton: By.css("[data-testid='button'][label='Video']"),
+  toggleSpeakerButton: By.css("[data-testid='button'][label='Speaker']"),
+  toggleContentButton: By.css("[data-testid='button'][label='Content']"),
+  roster: By.css(".roster"),
+  rosterCell: By.css("[data-testid='roster-cell']"),
+  hookStates: By.css("pre"),
+  attendeeState: By.css("code"),
+};
+
+
+class RosterTestPage extends BaseTestPage {
+  constructor(driver) {
+    super(driver);
+  }
+
+  async getRosterLength() {
+    const rosterCells = await this.findAll(ELEMENTS.rosterCell);
+    return rosterCells.length;
+  }
+
+  async getAttendeeRosterCell(attendeeName) {
+    const rosterCells = await this.findAll(ELEMENTS.rosterCell);
+    for (const rosterCell of rosterCells) {
+      const name = await rosterCell.getText();
+      if (name === attendeeName) {
+        return rosterCell;
+      }
+    }
+  }
+
+  async getAttendeeStates(attendeeName) {
+    const hookStates = this.find(ELEMENTS.hookStates);
+    const attendeeStates = await hookStates.findElements(ELEMENTS.attendeeState);
+
+    for (const attendeeState of attendeeStates) {
+      const testId = await attendeeState.getAttribute('data-testid');
+      if (testId === ('code-' + attendeeName)) {
+        return JSON.parse(await attendeeState.getText());
+      }
+    }
+
+    return {};
+  }
+
+  async getAttendeeState(attendeeName, property) {
+    const attendeeState = await this.getAttendeeStates(attendeeName);
+    const hookState = Object.values(attendeeState)[0];
+    return hookState[property];
+  }
+
+  async getLengthOfAttendeeStates() {
+    const hookStates = this.find(ELEMENTS.hookStates);
+    const attendeeStates = await hookStates.findElements(ELEMENTS.attendeeState);
+    return attendeeStates.length;
+  }
+
+  async checkRosterLength(targetLength) {
```

Comment:
So for all the `check..` APIs, do we want to name them as `assetRosterLength..`? Currently, it is not clear what this returns or if fails or asserts, if we are going to `asset` inside these `check` methods I think `assertRosterLength` makes it clear IMO. Do you have any early discussion or what do you think?

Reply:
I think `checkXxxxx` should be clear enough. The duty of these APIs is to check if the conditions are met. Their return type is void.

For example, `checkRosterLength(targetLength)` means to check if the length of roster is equal to `targetLength`. It might be clearer to name them as `checkIfRosterLengthEqualsTo(targetLength)`, but it's too redundant in my opinion. Sometimes we need to check if something is not equals to some value as well.

In JS SDK integration testing, we call these steps as `xxxxCheck` and we put them under the `integration/js/checks` folder.

Reply:
Hmm yeah fine to go ahead with it, just wanted to have some discussion over it. But just `check` was not clear enough until that assert line comes. It does not inherently tell what is returned when the failure happens whether it returns a boolean of check passed or failed. The action still remains unclear when one is not familiar with the code and wants to use it. 

Hope this article makes it clear (our discussion is between Assert and Check which is close though :) ): https://artoftesting.com/difference-between-assert-and-verify